src/lib/core/cbus.h

```diff
@@ -217,12 +217,11 @@ cpipe_submit_flush(struct cpipe *pipe)
 static inline void
 cpipe_push(struct cpipe *pipe, struct cmsg *msg)
 {
-	assert(loop() == pipe->producer);
```

Comment:
AFAICS the producer can be either `loop()` or `NULL` - we never set any other value for it. So I wouldn't let the user pass an arbitrary loop. Instead, I'd introduce a new function for creating a "loop-less" pipe (`cpipe_create_noev`?). IMO that would be more explicit. Then we wouldn't need to patch all the places where `cpipe_create` is called. I'd also leave this assertion be.

Reply:
I can't, unfortunately. In the tests I have a worker thread with `loop() != NULL` which creates a ev-less pipe having `producer == NULL`.